### PR TITLE
KAFKA-17021: Migrate AclCommandTest to new test infra

### PR DESCRIPTION
**Pull request by wernerdv**
Migrated AclCommandTest to new test infra.
Also rewritten it by java.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/admin/AclCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+                    asScalaSet(Collections.emptySet()),
+                    authorizer,
+                    new ResourcePattern(TOPIC, "Test-", PREFIXED),
+                    ANY
+            );
+        });
+    }
+
+    private static Map<Set<ResourcePattern>, Set<AccessControlEntry>> producerResourceToAcls(boolean enableIdempotence) {
+        return new HashMap<Set<ResourcePattern>, Set<AccessControlEntry>>() {{
```
Comment (chia7712) [inline core/src/test/java/kafka/admin/AclCommandTest.java]: Could you please use following code to replace it?
```java
    private static Map<Set<ResourcePattern>, Set<AccessControlEntry>> producerResourceToAcls(boolean enableIdempotence) {
        Map<Set<ResourcePattern>, Set<AccessControlEntry>> result = new HashMap<>();
        result.put(TOPIC_RESOURCES, asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW, asScalaSet(
                new HashSet<>(Arrays.asList(WRITE, DESCRIBE, CREATE))), asScalaSet(HOSTS))));
        result.put(TRANSACTIONAL_ID_RESOURCES, asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW, asScalaSet(
                new HashSet<>(Arrays.asList(WRITE, DESCRIBE))), asScalaSet(HOSTS))));
        result.put(Collections.singleton(CLUSTER_RESOURCE), asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW,
                enableIdempotence ? asScalaSet(Collections.singleton(IDEMPOTENT_WRITE)) : asScalaSet(Collections.emptySet()), asScalaSet(HOSTS))));
        return result;
    }
```

File: core/src/test/java/kafka/admin/AclCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    private static final Map<Set<ResourcePattern>, List<String>> RESOURCE_TO_COMMAND = new HashMap<Set<ResourcePattern>, List<String>>() {{
+            put(TOPIC_RESOURCES, Arrays.asList("--topic", "test-1", "--topic", "test-2"));
+            put(Collections.singleton(CLUSTER_RESOURCE), Collections.singletonList("--cluster"));
+            put(GROUP_RESOURCES, Arrays.asList("--group", "testGroup-1", "--group", "testGroup-2"));
+            put(TRANSACTIONAL_ID_RESOURCES, Arrays.asList("--transactional-id", "t0", "--transactional-id", "t1"));
+            put(TOKEN_RESOURCES, Arrays.asList("--delegation-token", "token1", "--delegation-token", "token2"));
+            put(USER_RESOURCES, Arrays.asList("--user-principal", "User:test-user1", "--user-principal", "User:test-user2"));
+        }};
+
+    private static final Map<Set<ResourcePattern>, Tuple2<Set<AclOperation>, List<String>>> RESOURCE_TO_OPERATIONS =
```
Comment (chia7712) [inline core/src/test/java/kafka/admin/AclCommandTest.java]: Could you please use `Entry` to replace ``Tuple2``?

**Review by chia7712 — COMMENTED** [review]
@wernerdv thanks for this patch. Could you please add unit test for `checkArgs` also? noted that you can add `@Test` to mark the test case as UT which can run without embedded kafak. Also, please move the `ClusterInstance` from constructor to test case. Otherwise, you will encounter the error when you mix together UT and IT.

File: core/src/test/java/kafka/admin/AclCommandTest.java (hunk 1 above)
Comment (wernerdv) [reply core/src/test/java/kafka/admin/AclCommandTest.java]: Done

File: core/src/test/java/kafka/admin/AclCommandTest.java (hunk 2 above)
Comment (wernerdv) [reply core/src/test/java/kafka/admin/AclCommandTest.java]: Done

**Comment by wernerdv** [comment]
@chia7712 Hello, sorry for the delay. 
Added unit tests for `checkArgs`, PR is ready for review.

File: core/src/test/java/kafka/admin/AclCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+            fullCmd.addAll(Arrays.asList(principalCmd, user.toString()));
+        }
+
+        return fullCmd;
+    }
+
+    private void withAuthorizer(ClusterInstance cluster, Consumer<Authorizer> consumer) {
+        if (cluster.isKRaftTest()) {
+            List<Authorizer> allAuthorizers = new ArrayList<>();
+            allAuthorizers.addAll(asKraftCluster(cluster).brokers().values().stream()
```
Comment (chia7712) [inline core/src/test/java/kafka/admin/AclCommandTest.java]: Do we really need `asKraftCluster` ? it seems to me `ClusterInstance#brokers` and  `ClusterInstance#controllers` should work well in this case.

File: core/src/test/java/kafka/admin/AclCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,729 @@
...
+                    .map(server -> server.authorizer().get()).collect(Collectors.toList()));
+            allAuthorizers.addAll(asKraftCluster(cluster).controllers().values().stream()
+                    .map(server -> server.authorizer().get()).collect(Collectors.toList()));
+            allAuthorizers.forEach(consumer);
+        } else {
+            Properties props = new Properties();
+            props.putAll(cluster.config().serverProperties());
+            props.put("zookeeper.connect", zkConnect(cluster));
+            try (AclAuthorizer auth = new AclAuthorizer()) {
+                auth.configure(KafkaConfig.fromProps(props, false).originals());
```
Comment (chia7712) [inline core/src/test/java/kafka/admin/AclCommandTest.java]: Could we pick up one of zk broker to reuse the created `AclAuthorizer`?

**Review by chia7712 — COMMENTED** [review]
@wernerdv thanks for updated patch. overall LGTM.

BTW, #16601 will remove the `LogCaptureAppender`. That will cause conflicts to this PR

**Comment by chia7712** [comment]
@wernerdv could you please fix the conflicts? thanks!

File: core/src/test/java/kafka/admin/AclCommandTest.java (hunk 3 above)
Comment (wernerdv) [reply core/src/test/java/kafka/admin/AclCommandTest.java]: Done

File: core/src/test/java/kafka/admin/AclCommandTest.java (hunk 4 above)
Comment (wernerdv) [reply core/src/test/java/kafka/admin/AclCommandTest.java]: Done

**Comment by wernerdv** [comment]
@chia7712 all conflicts are resolved.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**